core/test/dataset_test_common.cpp

```diff
@@ -4,6 +4,30 @@
 
 #include "dataset_test_common.h"
 
+std::vector<bool> make_bools(const scipp::index size,
+                             std::initializer_list<bool> pattern) {
+
+  using T = std::decay_t<decltype(pattern)>;
+  struct cycle_over {
+    T m_sequence;
+    typename T::iterator m_current;
+    cycle_over(T sequence) : m_sequence(sequence) {
+      m_current = m_sequence.begin();
+    };
+    typename T::value_type operator()() {
+      if (m_current == m_sequence.end())
+        m_current = m_sequence.begin();
+      return *(m_current++);
+    }
+  };
+
+  std::vector<bool> result(size);
+  std::generate(result.begin(), result.end(), cycle_over(pattern));
```

Comment:
Isn't this a case where a plain loop is much briefer?

```cpp
std::vector<bool> result(size);
auto it = pattern.begin();
for (auto &x : result) {
  if(it == pattern.end())
    it = pattern.begin();
  x = *(it++);
}
```

Reply:
Thanks. I've mainly taken on your suggestion. Annoyingly for `std::vector<bool>` `auto &` cannot be used because cannot bind mutable ref to temporary (result of specific container operator dereference) so my solution looks very slightly different.